crates/bevy_math/src/bounding/raytest2d.rs

```diff
@@ -0,0 +1,78 @@
+use super::{Aabb2d, BoundingCircle, IntersectsVolume};
+use crate::{primitives::Direction2d, Vec2};
+
+/// A raycast intersection test for 2D bounding volumes
+pub struct RayTest2d {
+    /// The origin of the ray
+    pub origin: Vec2,
+    /// The direction of the ray
+    pub dir: Direction2d,
+    /// The inverse direction of the ray
+    pub inv_dir: Vec2,
```

Comment:
Should these fields be public, given that they have to obey the invariant of inv_dir being the reciprocal of dir?